lib/virtio_mmio/virtio_rng.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2022 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * $FreeBSD$
+ */
+
+#include <openamp/open_amp.h>
+#include <openamp/virtqueue.h>
+#include <openamp/virtio.h>
+#include <openamp/virtio_mmio.h>
```

Comment:
I would expect to not have this dependency and to use the services independently of the transport layer.

if not simple to do it, I propose As a first step coule be acceptable to have 
```
#ifndef (WITH_VIRTIO_MMIO)
        #error only VIRTIO-MMIO transport layer  supported
#endif
```

Reply:
Done.